dotCMS/src/main/java/com/dotmarketing/business/IdentifierFactoryImpl.java

```diff
@@ -206,26 +176,45 @@ protected Identifier findByURI(String siteId, String uri) throws DotHibernateExc
 	}
 
 	@Override
-	protected List<Identifier> findByParentPath(String siteId, String parent_path) throws DotHibernateException {
+	protected List<Identifier> findByParentPath(final String siteId, String parent_path) throws DotDataException {
 	    if(!parent_path.endsWith("/")) {
 	        parent_path=parent_path+"/";
 	    }
 	    parent_path = parent_path.toLowerCase();
 
-        HibernateUtil dh = new HibernateUtil(Identifier.class);
-        dh.setQuery("from identifier in class com.dotmarketing.beans.Identifier where parent_path = ? and host_inode = ?");
-        dh.setParam(parent_path);
-        dh.setParam(siteId);
-        return (List<Identifier>) dh.list();
-    }
+		DotConnect dc = new DotConnect();
+		dc.setSQL("select * from identifier where parent_path = ? and host_inode = ?");
+		dc.addParam(parent_path);
+		dc.addParam(siteId);
+		try {
+			return ConvertToPOJOUtil.convertDotConnectMapToIdentifier(dc.loadResults());
+		} catch (ParseException e) {
+			throw new DotDataException(e);
+		}
+	}
 
 	@Override
 	protected Identifier loadFromDb(String identifier) throws DotDataException, DotStateException {
 		if (identifier == null) {
 			throw new DotStateException("identifier is null");
 		}
-		HibernateUtil hu = new HibernateUtil(Identifier.class);
-		return (Identifier) hu.load(identifier);
+
+		DotConnect dc = new DotConnect();
+		dc.setSQL("select * from identifier where id = ?");
+		dc.addParam(identifier);
+
+		List<Identifier> results = null;
+		try {
+			results = ConvertToPOJOUtil.convertDotConnectMapToIdentifier(dc.loadResults());
+		} catch (ParseException e) {
+			throw new DotDataException(e);
+		}
+
+		if (results != null && results.size() > 0){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)